### ISSUE TITLE
boto3 update broke pynamodb

### ISSUE THREAD

**Issue by igoreisenberg** [pynamodb/PynamoDB#1082]
Hi!
We're using https://pypi.org/project/pynamodb/5.2.1/
All was good until today's deploy - we got this error stack when trying to invoke `get` on one of the pynamodb's Model

```
File /var/task/pynamodb/models.py:540 in get
data = cls._get_connection().get_item(
File /var/task/pynamodb/connection/table.py:213 in get_item
return self.connection.get_item(
File /var/task/pynamodb/connection/base.py:1197 in get_item
operation_kwargs = self.get_operation_kwargs(
File /var/task/pynamodb/connection/base.py:903 in get_operation_kwargs
operation_kwargs.update(self.get_identifier_map(table_name, hash_key, range_key, key=key))
File /var/task/pynamodb/connection/base.py:820 in get_identifier_map
tbl = self.get_meta_table(table_name)
File /var/task/pynamodb/connection/base.py:548 in get_meta_table
data = self.dispatch(DESCRIBE_TABLE, operation_kwargs)
File /var/task/pynamodb/connection/base.py:329 in dispatch
data = self._make_api_call(operation_name, operation_kwargs, settings)
File /opt/python/wrapt/wrappers.py:644 in __call__
return self._self_wrapper(self.__wrapped__, self._self_instance,
File /opt/python/epsagon/modules/pynamodb.py:43 in _wrapper
return wrapper(PynamoDBEventAdapter, wrapped, instance, args, kwargs)
File /opt/python/epsagon/modules/general_wrapper.py:28 in wrapper
response = wrapped(*args, **kwargs)
File /var/task/pynamodb/connection/base.py:358 in _make_api_call args locals
request_dict = self.client._convert_to_request_dict(
```

We've investigated and didnt found any changes in the pynamodb itself, but we've found that the requirements of the pynamodb project are open-ended:
`botocore>=1.12.54`

so as soon as botocore got updated from 1.24.96 to 1.25.0, pynamodb started to take it, since we didnt explicitly specified transitional dependencies.

Specifying usage of botocore==1.24.96 solved the issue (same with specifying previous version of boto3).
Was anything broken with that version update?

**Comment by tom-a-c** [comment]
Please see https://github.com/pynamodb/PynamoDB/issues/1081

**Comment by ikonst** [comment]
@igoreisenberg I believe the change was part of [botocore 1.28](https://github.com/boto/botocore/blob/develop/CHANGELOG.rst#1280), not 1.25.

**Comment by ikonst** [comment]
This should be fixed in 5.2.3.
